main_test.go

```diff
@@ -360,3 +361,46 @@ func TestAlises(t *testing.T) {
 		}
 	}
 }
+
+func TestParseIniFile(t *testing.T) {
+	configHomeEnv := "XDG_CONFIG_HOME"
+	originalConfigHome := os.Getenv(configHomeEnv)
+	defer os.Setenv(configHomeEnv, originalConfigHome)
+
+	tests := []struct {
+		message       string
+		configPath    string
+		expectedError string
+	}{
+		{
+			message:       "valid ini file should parse correctly",
+			configPath:    path.Join("testdata", "ini", "valid"),
+			expectedError: "",
+		},
+		{
+			message:       "absent ini file should be ignored",
+			configPath:    path.Join("testdata", "ini", "missing"),
+			expectedError: "",
+		},
+		{
+			message:       "invalid ini file should raise error",
+			configPath:    path.Join("testdata", "ini", "invalid"),
+			expectedError: "couldn't read \"testdata/ini/invalid/yab/yab.ini\": \"testdata/ini/invalid/yab/yab.ini:2: time: unknown unit foo in duration 3foo\"",
+		},
+	}
+
+	for _, tt := range tests {
+		os.Setenv(configHomeEnv, tt.configPath)
+
+		parser, _ := newParser()
+
+		err := parseDefaultConfigs(parser)
+
+		if tt.expectedError == "" {
+			assert.Nil(t, err, tt.message)
+		} else {
+			assert.NotNil(t, err, tt.message)
+			assert.Equal(t, err.Error(), tt.expectedError, tt.message)
```

Comment:
Make this check happen only if the previous check works, or this will panic.

```go
if assert.Error(t, err, tt.message) {
    assert.Equal(t, err.Error(), tt.expectedError, tt.message)
}
```